### PR TITLE
Fix potential exponential backtracking when removing comments

### PR DESCRIPTION
**Pull request by jlelong**
Related to CodeQL inefficient regular expression warning.

### CODE REVIEW

**Comment by tamuratak** [comment]
I think the function has a bug. It does not remove a comment immediately after `\\`.

https://github.com/James-Yu/LaTeX-Workshop/blob/54fa70f22910d0fe50e991d919e4845c63694db7/src/utils/utils.ts#L29-L33

```latex
\begin{align*}
x \\%abc
y
\end{align*}
```

**Comment by jlelong** [comment]
Indeed! This bug has been around for two years and nobody ever complained.

**Comment by jlelong** [comment]
Not yet!

File: src/utils/utils.ts [hunk 1]
```diff
@@ -26,8 +26,8 @@ export function escapeRegExp(str: string) {
  * @return the input text with comments removed.
  * Note the number lines of the output matches the input
  */
-export function stripComments(text: string, commentSign: string): string {
-    const pattern = '([^\\\\]|^)' + commentSign + '.*$'
+export function stripComments(text: string, commentSign: string = '%'): string {
+    const pattern = '(^|[^\\\\]|(?:(?<!\\\\)\\\\\\\\))' + commentSign + '.*$'
```
Comment (tamuratak) [inline src/utils/utils.ts:30]: I don't like the API. It makes things too complicated unnecessarily. Please delete the `commentSign` argument and use a literal regular expression. When we need to remove comments which begin other than `%`, let's define another function.

**Review by tamuratak — CHANGES_REQUESTED** [review]

**Comment by tamuratak** [comment]
It does not remove the following comment.

```latex
\begin{align*}
x \\\\%abc
y
\end{align*}
```

**Comment by jlelong** [comment]
This was done on purpose because I thought two consecutive newlines `\\` were not valid LaTeX. Although, it is not inside `align`, it is in text mode. I will fix it.

**Comment by jlelong** [comment]
It should be fine now.

**Review by tamuratak — APPROVED** [review]

**Pull request merged**